src/graphapi/matmul.cpp

```diff
@@ -104,5 +104,254 @@ void BackendMatmulDescriptor::getAttribute(miopenBackendAttributeName_t attribut
     }
 }
 
+OperationMatmul OperationMatmulBuilder::build()
+{
+    if(!aSet || !bSet || !cSet || !matmulSet)
+        MIOPEN_THROW(miopenStatusBadParm);
+
+    if(!(mOperationMatmul.mA->getDimensions().size() == mOperationMatmul.mB->getDimensions().size() == mOperationMatmul.mC->getDimensions().size()))
+        MIOPEN_THROW(miopenStatusBadParm);
+    int Am = mOperationMatmul.mA->getDimensions().end()[-2];
```

Comment:
[Q] Do we guarantee that `mOperationMatmul.mA->getDimensions().end()[-2]` exists?

Reply:
Fixed in setters, added tests

Reply:
[resolved]

Reply:
```suggestion
    int Am = mOperationMatmul.mA->getDimensions().begin()[aDimensionsCount-2];
```
`cppcheck` doesn't allow to dereference `end()` iterators.